src/content_script/mastodonInject.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @typedef {Object} VersionNumber
+ * @property {string} major
+ * @property {string} minor
+ * @property {string} patch
+ */
+
+/**
+ * parses a versionNumber string
+ * @returns {VersionNumber}
+ */
+function parseVersionNumber() {
+  const versionElement = document.querySelector(`.link-footer`).textContent
+  const versionNumber = versionElement.match(/v?(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)$/)
+  return versionNumber.groups
+}
+
+
+/**
+ * Replacement onClick handler for Follow button
+ * @param {Event} event 
+ */
+function onClickFollow (event) {
+  event.stopPropagation()
+  event.preventDefault()
+  const username = window.location.pathname.split(`/`).slice(-1)[0]
+  // activate AutoRemoteFollow
+  window.open(`/users/${username}/remote_follow`, `_blank`)
+}
+
+/**
+ * wait for element to appear
+ * @param {string} selector
+ * @param {number} timeout
+ */
+function waitForElement (selector, timeout) {
+  return new Promise((resolve, reject) => {
+    function getElement() {
+      return document.querySelector(selector)
+    }
+
+    const element = getElement()
+    if(element){
+      return resolve(element)
+    }
+
+    const observer = new MutationObserver(() => {
+      const element = getElement()
+      if(element){
+        resolve(element)
+        observer.disconnect()
+      }
+    })
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true
+    })
+
+    window.setTimeout(() => {
+      reject()
+    }, timeout)
+  }) 
+}
+
+/**
+ * Inject replacement onClick handler for Follow button
+ */
+async function injectFollowButton () {
+  try {
+    const followButton = await waitForElement(`.account__header__tabs__buttons button.button`, 20000)
+    followButton.addEventListener(`click`, onClickFollow)
+  } catch {
+    // Follow button failed to appear
+  }
+}
+
+async function init () {
+  let versionNumber
+
+  try {
+    const initialStateObject = JSON.parse(document.getElementById(`initial-state`).innerHTML)
```

Comment:
Secondly, I very much like the idea and it seems wise to use this. However, is this an official "API" of Mastodon?

I again would like to avoid breakage in case of a update or in case a server admins decides to remove that version number?

Reply:
No, I don't think this is an official Mastodon API, but it is how much of Mastodon's frontend code gets the initial state from the backend.

I agree we should use something official if possible, but there doesn't appear to be any official way to get the version number.